File: src/models/alert-collection.ts
import { Alert } from './alert';
import { Type, Expose } from 'class-transformer';

export class AlertCollection {
	public title!: string;
	@Type(() => Date)
	public updated!: Date;
	@Type(() => Alert)
	@Expose({ name: '@graph' })
	public alerts!: Alert[];
}

File: WeatherGoat/Models/AlertCollection.cs
using System.Text.Json.Serialization;

namespace WeatherGoat.Models;

public record AlertCollection
{
    [JsonPropertyName("title")]
    public string Title { get; set; }
    
    [JsonPropertyName("updated")]
    public DateTime UpdatedAt { get; set; }
    
    [JsonPropertyName("@graph")]
    public IReadOnlyList<Alert> Alerts { get; set; }

    public bool HasAlerts => Alerts.Count > 0;
}
